test/functional/test-url.js

```diff
@@ -120,6 +117,16 @@ describe('parseUrl', () => {
     const a2 = parseUrl(url);
     expect(a1).to.equal(a2);
   });
+  it('caches up to 100 results', () => {
+    const url = 'https://foo.com:123/abc?123#foo';
+    const a1 = parseUrl(url);
+    // cache 99 more urls in order to reach max capacity of LRU cache: 100
+    for (let i = 0; i < 100; i++) {
+      parseUrl(`${url}-${i}`);
```

Comment:
expecting result of parseUrl to be same as a1 would be a nice addition to ensure cache part is working as well.

Reply:
@aghassemi Ah right, that definitely makes sense. I'll add that in